backend/src/map_model.rs

```diff
@@ -567,34 +622,94 @@ impl MapModel {
         Ok(())
     }
 
+    pub fn router_input_before(&self) -> impl RouterInput + use<'_> {
+        struct RouterInputBefore<'a> {
+            map: &'a MapModel,
+        }
+        impl RouterInput for RouterInputBefore<'_> {
+            fn roads_iter(&self) -> impl Iterator<Item = &Road> {
+                self.map.roads.iter()
+            }
+
+            fn get_r(&self, r: RoadID) -> &Road {
+                self.map.get_r(r)
+            }
+
+            fn get_i(&self, i: IntersectionID) -> &Intersection {
+                self.map.get_i(i)
+            }
+
+            fn modal_filter(&self, r: RoadID) -> Option<&ModalFilter> {
+                self.map.original_modal_filters.get(&r)
+            }
+
+            fn travel_flow(&self, r: RoadID) -> TravelFlow {
+                let road = self.get_r(r);
+                TravelFlow::from_osm(&road.tags)
+            }
+
+            fn diagonal_filter(&self, _i: IntersectionID) -> Option<&DiagonalFilter> {
+                // TODO: do we want to import pre-existing diagonal filters from OSM at some point?
```

Comment:
I think they'll always be expressed in a different way, that might be kind of hard for us to "correct" to a simple 4-way intersection. Roanoke park is the only one in immediate memory:
https://www.openstreetmap.org/way/51538523
![image](https://github.com/user-attachments/assets/4529d909-d1f0-475b-99b2-b880b1f9e170)
We'd have to detect this and stitch the two roads together somehow...

Reply:
I suppose this is something that could be mitigated somewhat by #40.